sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ColumnResolutionHelper.scala

```diff
@@ -539,4 +533,28 @@ trait ColumnResolutionHelper extends Logging {
         None
     }
   }
+
+  private def findPlanById(
```

Comment:
can we do tailrec?

Reply:
can't find a way to make it tailrec... I need to check the return value from `plan.children.flatMap(findPlanById(u, id, _))`